api/hasura/actions/_handlers/createVaultTx.ts

```diff
@@ -0,0 +1,69 @@
+import assert from 'assert';
+
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import {
+  ValueTypes,
+  vault_tx_types_enum,
+} from '../../../../api-lib/gql/__generated__/zeus';
+import { adminClient } from '../../../../api-lib/gql/adminClient';
+import { verifyHasuraRequestMiddleware } from '../../../../api-lib/validate';
+import {
+  VaultLogInputSchema,
+  VaultLogUnionSchema,
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+} from '../../../../src/lib/zod';
+
+async function handler(req: VercelRequest, res: VercelResponse) {
+  const {
+    session_variables: { hasuraProfileId },
+    input: { payload },
+  } = composeHasuraActionRequestBodyWithSession(
+    VaultLogInputSchema,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+  const actionToLog = VaultLogUnionSchema.parse(payload);
+  switch (actionToLog.tx_type) {
+    case 'Deposit': {
+      const result = await logVaultTx({
+        ...actionToLog,
+        created_by: hasuraProfileId,
+        tx_type: vault_tx_types_enum.Deposit,
```

Comment:
might be able to shorten the code if you do something like this
```ts
created_by: hasuraProfileId,
tx_type: (() => {
      switch (actionToLog.tx_type) {
        case 'Deposit':
          return vault_tx_types_enum.Deposit;
        case 'Withdraw':
          return vault_tx_types_enum.Withdraw;
        case 'Distribution':
          return vault_tx_types_enum.Distribution;
    }})()
```

Reply:
ah, this is one of the few places where I actually like the idea of using an IIFE. Thanks!